Update UG and DG for recent changes

lyueyang opened the pull request:
- As Program has been changed to accommodate a HR executive as the user instead of a passenger and driver, use case updated to reflect this change of focus.
- Full stops ended to each sentence
- MSS indicated

lyueyang committed:
Update DG to HR Executive as user

- As Program has been changed to accommodate a HR executive as the user instead of a passenger and driver, use case updated to reflect this change of focus.
- Full stops ended to each sentence
- MSS indicated

lyueyang committed:
Update UG

- Example name of driver changed to Alice to reflect female driver example
- Instances of drive changed to pool

lyueyang committed:
Change DG

- Update value proposition to reflect change of targeted user
- Change styling for table

ZechariahTan commented on docs/DeveloperGuide.md:
```diff
@@ -150,37 +150,36 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+|* * *   |HR Executive  |search for specific type of passengers                           |I can see if any passengers fulfil a criteria and view their carpool details                    |
+|* * *   |HR Executive  |list all passengers                                              |I can see all the passengers available                                                          |
+|* * *   |HR Executive  |create a profile                                                 |I can easily manage and track drivers and passengers                                            |
+|* * *   |HR Executive  |delete my profile                                                |passenger's data will not be stored when they are no longer looking to carpool                  |
```
I think you might have to 'change delete my profile to 'delete employee profile' as the user is the HR exec now and they're managing other employee's profiles.

lyueyang replied:
Good catch, let me amend that

ZechariahTan reviewed (approved):
LGTM

lyueyang committed:
Correct error on user stories

- One of the user requirements were not properly reflected in the user stories table to show HR executive as the user

lyueyang committed:
Resolve master branch conflicts

ZechariahTan reviewed (approved):
LGTM
